ssas/service/main/main.go

```diff
@@ -52,86 +52,97 @@ import (
+func main() {
+	ssas.Logger.Info("Home of the System-to-System Authentication Service")
+	var config = parseConfig()
+	handleFlags(config)
+}
+
+func parseConfig() Flags {
+	var flags Flags
 	const usageAddFixtureData = "unconditionally add fixture data"
-	flag.BoolVar(&doAddFixtureData, "add-fixture-data", false, usageAddFixtureData)
+	flag.BoolVar(&flags.doAddFixtureData, "add-fixture-data", false, usageAddFixtureData)
 
 	const usageResetSecret = "reset system secret for the given client_id; requires client-id flag with argument"
-	flag.BoolVar(&doResetSecret, "reset-secret", false, usageResetSecret)
+	flag.BoolVar(&flags.doResetSecret, "reset-secret", false, usageResetSecret)
 
 	const usageNewAdminSystem = "add a new admin system to the service; requires system-name flag with argument"
-	flag.BoolVar(&doNewAdminSystem, "new-admin-system", false, usageNewAdminSystem)
-	flag.StringVar(&systemName, "system-name", "", "the system's name (e.g., 'BCDA Admin')")
+	flag.BoolVar(&flags.doNewAdminSystem, "new-admin-system", false, usageNewAdminSystem)
+	flag.StringVar(&flags.systemName, "system-name", "", "the system's name (e.g., 'BCDA Admin')")
 
 	const usageListIPs = "list all IP addresses registered to active systems"
-	flag.BoolVar(&doListIPs, "list-ips", false, usageListIPs)
+	flag.BoolVar(&flags.doListIPs, "list-ips", false, usageListIPs)
 
 	const usageListExpCreds = "list credentials about to expire or timeout due to inactivity"
-	flag.BoolVar(&doListExpCreds, "list-exp-creds", false, usageListExpCreds)
+	flag.BoolVar(&flags.doListExpCreds, "list-exp-creds", false, usageListExpCreds)
 
 	const usageShowXData = "display group xdata"
-	flag.BoolVar(&doShowXData, "show-xdata", false, usageShowXData)
-	flag.StringVar(&auth, "auth", "", "an auth header containing the hashed client id")
+	flag.BoolVar(&flags.doShowXData, "show-xdata", false, usageShowXData)
+	flag.StringVar(&flags.auth, "auth", "", "an auth header containing the hashed client id")
 
 	const usageStart = "start the service"
-	flag.BoolVar(&doStart, "start", false, usageStart)
+	flag.BoolVar(&flags.doStart, "start", false, usageStart)
 
 	// used by both `doResetSecret` and `doGetXdata`
-	flag.StringVar(&clientID, "client-id", "", "a system's client id")
-
-	appName := os.Getenv("NEW_RELIC_APP_NAME")
-	licenseKey := os.Getenv("NEW_RELIC_LICENSE_KEY")
-	_, err := newrelic.NewApplication(
-		newrelic.ConfigAppName(appName),
-		newrelic.ConfigLicense(licenseKey),
-	)
-	if nil != err {
-		ssas.Logger.Warnf("New Relic integration is disabled: %s", err)
-	}
+	flag.StringVar(&flags.clientID, "client-id", "", "a system's client id")
+	flag.Parse()
+	return flags
 }
 
-// We provide some simple commands for bootstrapping the system into place. Commands cannot be combined.
-func main() {
-	ssas.Logger.Info("Home of the System-to-System Authentication Service")
-
-	flag.Parse()
-	if doAddFixtureData {
+func handleFlags(flags Flags) {
```

Comment:
are these flags set from environment variables?

Reply:
They're passed in directly when the compiled go program is run, so something like `bcda --add-fixture-data` 😄